test/test.js

```diff
@@ -465,7 +475,7 @@ suite('Body mixin', function() {
     })
 
     test('handles json parse error', function() {
-      return fetch('/json-error').then(function(response) {
```

Comment:
I'm pretty sure we still need to return a Promise to Mocha, to keep the test running until we're done. Can't we return a resolved promise after we've handled the (expected) failure?